consensus/bor/bor.go

```diff
@@ -774,23 +781,64 @@ func (c *Bor) Seal(chain consensus.ChainReader, block *types.Block, results chan
 
 	// Wait until sealing is terminated or delay timeout.
 	log.Trace("Waiting for slot to sign and propagate", "delay", common.PrettyDuration(delay))
+	shouldSeal := make(chan bool)
+	go c.WaitForSealingOp(number, shouldSeal, delay, stop)
 	go func() {
-		select {
-		case <-stop:
+		defer func() {
+			close(shouldSeal)
+			c.activeSealingOp = nil
+		}()
+		switch <-shouldSeal {
+		case false:
 			return
-		case <-time.After(delay):
+		case true:
+			if wiggle > 0 {
+				log.Info(
+					"Sealing out-of-turn",
+					"number", number,
+					"wiggle", common.PrettyDuration(wiggle),
+					"in-turn-signer", snap.ValidatorSet.GetProposer().Address.Hex(),
+				)
+			}
+			log.Info(
+				"Sealing successful",
+				"number", number,
+				"delay", delay,
+				"headerDifficulty", header.Difficulty,
+			)
 		}
 
 		select {
 		case results <- block.WithSeal(header):
 		default:
-			log.Warn("Sealing result is not read by miner", "sealhash", SealHash(header))
+			log.Warn("Sealing result was not read by miner", "sealhash", SealHash(header))
 		}
 	}()
-
 	return nil
 }
 
+// WaitForSealingOp blocks until delay elapses or stop signal is received
+func (c *Bor) WaitForSealingOp(number uint64, shouldSeal chan bool, delay time.Duration, stop <-chan struct{}) {
+	ctx, cancel := context.WithCancel(context.Background())
+	c.activeSealingOp = &ActiveSealingOp{number, cancel}
+	select {
+	case <-stop:
+		shouldSeal <- false
+	case <-ctx.Done():
+		shouldSeal <- false
+	case <-time.After(delay):
+		shouldSeal <- true
+	}
+}
+
+// CancelActiveSealingOp cancels in-flight sealing process
+func (c *Bor) CancelActiveSealingOp() {
+	if c.activeSealingOp != nil {
+		log.Info("Discarding active sealing operation", "number", c.activeSealingOp.number)
```

Comment:
Should this be a debug statement instead?

Reply:
Yes

Reply:
Done.